Kernel/Bus/VirtIO/Console.cpp

```diff
@@ -12,14 +12,20 @@ namespace Kernel::VirtIO {
 
 unsigned Console::next_device_id = 0;
 
-UNMAP_AFTER_INIT NonnullRefPtr<Console> Console::must_create(PCI::Address address)
+UNMAP_AFTER_INIT RefPtr<Console> Console::try_create(PCI::Address address)
 {
-    return adopt_ref_if_nonnull(new Console(address)).release_nonnull();
+    auto console = adopt_ref_if_nonnull(new Console(address));
+    if (!console)
+        return {};
+    if (!console->initialize())
+        return {};
+    return console;
 }
 
-UNMAP_AFTER_INIT void Console::initialize()
+UNMAP_AFTER_INIT bool Console::initialize()
 {
-    Device::initialize();
+    if (auto result = Device::initialize(); !result)
+        return false;
```

Comment:
It would be nice to switch to the return type of `::initialize()` to KResult, so we could leverage `TRY` here

Reply:
Well, I don't think this is a good idea. The reason for this is that we don't really propagate this back to userspace (remember, this is not something that is done because the user requested it), so what kind of error should we support and return? only `ENOMEM` is valid, but all other errors as far as I can tell, have a userspace meaning (for example, `ENOENT` is for when you don't find the directory entry when you try to find or delete a directory entry).
This is why I don't think `KResult` is appropriate here, maybe `Result` is a better option (so we still have a strong typed return value).

Reply:
AFAIK, we use ENOMEM-only KResult in other places too, but up to you.

Reply:
I applied the fix in another commit, to make it more bisectable :)